@app/client/src/pages/api/s3.tsx

```diff
@@ -0,0 +1,67 @@
+import { NextApiRequest, NextApiResponse } from "next";
+import AWS from "aws-sdk";
+import getConfig from "next/config";
+const { serverRuntimeConfig } = getConfig();
+
+export default async (req: NextApiRequest, res: NextApiResponse) => {
+  const bucket = serverRuntimeConfig.BUCKET;
+  const key = req.query.key;
+  const params: AWS.S3.PutObjectRequest = {
+    Bucket: bucket,
+    Key: key as string,
+  };
+  const client = getClient();
+  const operation = req.query.operation;
+  if (operation === "put") {
+    put(client, params);
+  } else if (operation === "delete") {
+    del(client, params);
+  }
+
+  function getClient() {
+    const region = serverRuntimeConfig.AWS_REGION;
+    const accessKey = serverRuntimeConfig.AWSACCESSKEYID;
+    const secretKey = serverRuntimeConfig.AWSSECRETKEY;
+    AWS.config.update({
+      accessKeyId: accessKey,
+      secretAccessKey: secretKey,
+      signatureVersion: "v4",
+      region: region,
+    });
+    const options = {
+      signatureVersion: "v4",
+      region: region,
+      // uncomment to use accelerated endpoint
+      // accelerated endpoint must be turned on in your s3 bucket first
+      // endpoint: new AWS.Endpoint(
+      //   "bucket.s3-accelerate.amazonaws.com"
+      // ),
+      // useAccelerateEndpoint: true,
+    };
+    const client = new AWS.S3(options);
+    return client;
+  }
+  function put(client: AWS.S3, params: AWS.S3.PutObjectRequest) {
+    const putParams = {
+      ...params,
+      Expires: 5 * 60,
+    };
+
+    client.getSignedUrl("putObject", putParams, (err, url) => {
+      if (err) {
+        res.json({ success: false, err });
+      } else {
+        res.json({ success: true, url });
```

Comment:
Rather than using a `success` key in the JSON payload, it's better to use the HTTP status code to communicate an error. In this case, if the error is coming from S3, that indicates the status "502 Bad Gateway" because our server is acting as a gateway (or proxy) to S3.

```js
if (err) {
  res.status(504).json({ err });
} else {
  res.json({ url });
}
```